src/functions.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+function connectToDB(string $dbName): PDO {
+    $connectionString ="mysql:host=db; dbname=$dbName";
+    $db = new PDO($connectionString, 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+function getAllDreams(PDO $db): array {
+    $query = $db->prepare('SELECT * FROM dreams;');
+    $query->execute();
+    $dreams =  $query->fetchALL();
+    return $dreams;
+}
+
+function displayDreamDetails(array $dreams): string{
+    $dreamOutput ='';
+    foreach($dreams as $dream){
+        $dreamOutput .="<h3>" . $dream['dream_title'] . "</h3>
```

Comment:
what does the dreamOutput .= do? just haven't seen this before so intrigued

Reply:
left, as discussed!